p2p/src/peer_manager/mod.rs

```diff
@@ -1662,13 +1751,19 @@ where
     }
 }
 
+pub fn stale_tip_time_diff(chain_config: &ChainConfig) -> Duration {
+    // Note: bitcoin core also uses "3 * block_spacing" for stale tip detection, but their
+    // "block spacing" is 10 min instead of out 2. TODO: should we use bigger time diff?
```

Comment:
Yes. It should be like 30 minutes to 1 hour. Block height doesn't mean anything on its own for security. What matters for security is time (which translates to chain-trust).

Reply:
I made it 30 min (it's now in `PeerManagerConfig::stale_tip_time_diff`)